### PR TITLE
Add Badge component.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue #3810 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by johnPhillips** [comment]
@googlebot I consent.

File: assets/js/components/Badge.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+const Badge = ( { label } ) => (
+	<span className="googlesitekit-badge">{ label }</span>
```
Comment (tofumatt) [inline assets/js/components/Badge.js:25]: The IB mentions using `googlesitekit__badge`, but I think `googlesitekit-badge` _is_ the more appropriate CSS class name. Just flagging that here that it's been noticed but I think this is better. 🙂

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA.js [hunk 2]
```diff
@@ -90,14 +93,6 @@ function DashboardCTA( { Widget, WidgetNull } ) {
 		}
 	}, [ activateModule, navigateTo, setInternalServerError ] );
 
-	const onLinkClick = useCallback( () => {
-		trackEvent(
-			IDEA_HUB_GA_CATEGORY_WIDGET,
-			'click_outgoing_link',
-			'idea_hub_learn_more'
-		);
```
Comment (tofumatt) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA.js:98]: Why was this removed? I don't see it mentioned in the ACs/IB, and it seems like something worth tracking—to see if users are interested in this new feature.

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks like a tracking event was removed and I'm not sure why, but otherwise this looks good. The badges everywhere seem right to me 👍🏻

**Review by tofumatt — APPROVED** [review]

**Comment by tofumatt** [comment]
@googlebot I consent.

**Pull request merged**